### PR TITLE
Moved SQLite row identifier handling into separate methods

### PR DESCRIPTION
**Pull request by joachimmetz**
Moved SQLite row identifier handling into separate methods

### CODE REVIEW

File: plaso/storage/sqlite/sqlite_file.py [hunk 1]
```diff
@@ -431,6 +431,78 @@ def _ReadAndCheckStorageMetadata(self, check_readable_only=False):
...
+    Args:
+      event (EventObject):
+    """
+    row_identifier = getattr(event, 'event_data_row_identifier', None)
+    if row_identifier is not None:
+      event_data_identifier = identifiers.SQLTableIdentifier(
+          'event_data', row_identifier)
+      event.SetEventDataIdentifier(event_data_identifier)
+
+      delattr(event, 'event_data_row_identifier')
```
Comment (Onager) [inline plaso/storage/sqlite/sqlite_file.py:446]: Shouldn't this set to None, not delete the attribute? event_data_row_identifier part of the public api for the event container. This goes for the other methods too.

File: plaso/storage/sqlite/sqlite_file.py [hunk 2]
```diff
@@ -431,6 +431,78 @@ def _ReadAndCheckStorageMetadata(self, check_readable_only=False):
...
+    Args:
+      event (EventObject):
+
+    Raises:
+      IOError: if the event data identifier type is not supported.
+      OSError: if the event data identifier type is not supported.
+    """
+    event_data_identifier = event.GetEventDataIdentifier()
+
+    # TODO: change to no longer allow event_data_identifier is None
```
Comment (Onager) [inline plaso/storage/sqlite/sqlite_file.py:460]: Is this todo still required? I thought all parsers were setting event data now?

File: plaso/storage/sqlite/sqlite_file.py [hunk 3]
```diff
@@ -431,6 +431,78 @@ def _ReadAndCheckStorageMetadata(self, check_readable_only=False):
...
+    row_identifier = getattr(event, 'event_data_row_identifier', None)
+    if row_identifier is not None:
+      event_data_identifier = identifiers.SQLTableIdentifier(
+          'event_data', row_identifier)
+      event.SetEventDataIdentifier(event_data_identifier)
+
+      delattr(event, 'event_data_row_identifier')
+
+  def _UpdateEventDataIdentifierBeforeSerialize(self, event):
+    """Updates the event data identifier of an event before serialization.
```
Comment (Onager) [inline plaso/storage/sqlite/sqlite_file.py:449]: This doesn't update update event_data_identifier, it sets the event_data_row_identifier on the event. Please update the docstring.

File: plaso/storage/sqlite/sqlite_file.py [hunk 4]
```diff
@@ -431,6 +431,78 @@ def _ReadAndCheckStorageMetadata(self, check_readable_only=False):
...
+      raise ValueError('Missing event row identifier attribute')
+
+    event_identifier = identifiers.SQLTableIdentifier(
+        self._CONTAINER_TYPE_EVENT, row_identifier)
+    event_tag.SetEventIdentifier(event_identifier)
+
+    delattr(event_tag, 'event_row_identifier')
+
+  def _UpdateEventIdentifierBeforeSerialize(self, event_tag):
+    """Updates the event identifier of an event before serialization.
```
Comment (Onager) [inline plaso/storage/sqlite/sqlite_file.py:490]: This doesn't update update event_data_identifier, it sets the event_data_row_identifier on the event tag. Please update the docstring.

File: plaso/storage/sqlite/sqlite_file.py [hunk 5]
```diff
@@ -431,6 +431,78 @@ def _ReadAndCheckStorageMetadata(self, check_readable_only=False):
...
+    if event_data_identifier:
+      if not isinstance(event_data_identifier, identifiers.SQLTableIdentifier):
+        raise IOError('Unsupported event data identifier type: {0!s}'.format(
+            type(event_data_identifier)))
+
+      setattr(event, 'event_data_row_identifier',
+              event_data_identifier.row_identifier)
+
+  def _UpdateEventIdentifierAfterDeserialize(self, event_tag):
+    """Updates the event identifier of an event after deserialization.
```
[the remark this answers is not in the document]
Comment (joachimmetz) [reply plaso/storage/sqlite/sqlite_file.py:471]: done

File: plaso/storage/sqlite/sqlite_file.py [hunk 6]
```diff
@@ -431,6 +431,78 @@ def _ReadAndCheckStorageMetadata(self, check_readable_only=False):
     self.serialization_format = metadata_values['serialization_format']
     self.storage_type = metadata_values['storage_type']
 
+  def _UpdateEventDataIdentifierAfterDeserialize(self, event):
+    """Updates the event data identifier of an event after deserialization.
+
+    Args:
+      event (EventObject):
```
[the remark this answers is not in the document]
Comment (joachimmetz) [reply plaso/storage/sqlite/sqlite_file.py:438]: done.

File: plaso/storage/sqlite/sqlite_file.py [hunk 7]
```diff
@@ -431,6 +431,78 @@ def _ReadAndCheckStorageMetadata(self, check_readable_only=False):
...
+          'event_data', row_identifier)
+      event.SetEventDataIdentifier(event_data_identifier)
+
+      delattr(event, 'event_data_row_identifier')
+
+  def _UpdateEventDataIdentifierBeforeSerialize(self, event):
+    """Updates the event data identifier of an event before serialization.
+
+    Args:
+      event (EventObject):
```
[the remark this answers is not in the document]
Comment (joachimmetz) [reply plaso/storage/sqlite/sqlite_file.py:452]: done.

File: plaso/storage/sqlite/sqlite_file.py (hunk 1 above)
Comment (joachimmetz) [reply plaso/storage/sqlite/sqlite_file.py:446]: I'm mimicking the current behavior with `del`. We can chat more about about these special attributes and how to best handle them in our conversation.

File: plaso/storage/sqlite/sqlite_file.py (hunk 2 above)
Comment (joachimmetz) [reply plaso/storage/sqlite/sqlite_file.py:460]: I'll address this in a later CL, wanted to keep it out of this one.

File: plaso/storage/sqlite/sqlite_file.py (hunk 3 above)
Comment (joachimmetz) [reply plaso/storage/sqlite/sqlite_file.py:449]: done

File: plaso/storage/sqlite/sqlite_file.py (hunk 4 above)
Comment (joachimmetz) [reply plaso/storage/sqlite/sqlite_file.py:490]: done

**Pull request merged**